### ISSUE TITLE
[BUG][Block 657887]  `Native execution error: Native execution error: Native execution error: DIRECTION\n`

### ISSUE THREAD

**Issue by pefontana** [lambdaclass/cairo_native#739]
When executing Block 657887 with starknet-replay, we get this error in certain transactions 
```
Native execution error: Native execution error: Native execution error: Native execution error: DIRECTION\n
```
This error doesn't occur when we execute the block with only the cairo VM

transactions hash that triggers the error:
* 0x548c30d3ee15d5180c1cde487a7c4a66efd40dc8c3bad1afe9f98069523dcd4
* 0xd4e73c48b38d8394593c15b5c142d6d38ff16acf774453ae9b0f20dfc32c82

The same happens in the Block 659392

Transactions hashes:
* 0x1b2c498c06c6c3354c74d09dbc38de295cde904a2ef9bf10f8222ef9381be60
* 0x72e347547fb46f5ed1e23d84d72173da035fc4ad0c481f9564ca08beea73254
* 0x50614b4a762d3fecce94a58c2f63ed58665c0521e88e19f27894ed6217e069
* 0x6915b7ef6685ca9bcddb2b3b5d0007931fb91f9a85be1c636b5581257ffee62
* 0x78871ae0bb797ede5a80cd3659783ccf420d763505f1947af075622fb490ec0
* 0x357bf5d29d466fe2a1f9562a0373eb4c7c42411097eae267ac969be90b048f4
* 0x4b65ef109102f365244ff4613bd4f20c8ded773e530bb3ce993e9867385ca81
* 0x12880ce50c519bb6e959cbe71ea2ebdc5034a13874a08515210a8ffbb868b2e
* 0x19fbfbd511a8d87fb198860c251cb9e7baacda90badede1b52ba766346bdb8
* 0x58219ec9e52df2938e9b6ef18d626c35e1fdc92878a397c4aaec9ca2179d0f8
* 0x4194eca24d65b45b68ece95c5cee43fce0e17db20992f3e8cc50e570d239356
* 0x1baaf8b0de1a7d1dc5b70164f2406eef71e4f33985c183646dc002b08465736
* 0x4364a9864a2c1673c435ceee099de3cf3f559e4b2387fef3b5c86231fd486f2
* 0x56270b23b24d0ddffd93989252519e64b0cd43e6f5df88a666ccf0c871890ac
* 0x41ee056fc3399413d5cf5876e312bacc8e955ed89fdc9b62e346a9b26e50432

It also happens in other blocks, you can check in the execution logs inside the starknet-replay folder (in the server), files block-NUMBER.md

**Comment by azteca1998** [comment]
PR #778 fixes this bug. Some of those hashes still fail with other errors, but they also fail using the VM, therefore it may not be related to native specifically, but rather in the supported transaction versions or other compatibility issues.

**Comment by FrancoGiachetta** [comment]
Update 

Same as #738

**Comment by pefontana** [comment]
Fixed with #787 and #763
